File: apis/src/main/java/com/beat/apis/schedule/facade/ScheduleFacade.java
package com.beat.apis.schedule.facade;

import org.springframework.stereotype.Service;

import com.beat.apis.schedule.application.ScheduleService;
import com.beat.apis.schedule.application.dto.request.TicketAvailabilityRequest;
import com.beat.apis.schedule.application.dto.response.TicketAvailabilityResponse;

import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class ScheduleFacade {
	private final ScheduleService scheduleService;

	public TicketAvailabilityResponse findTicketAvailability(Long scheduleId, int purchaseTicketCount) {
		return scheduleService.findTicketAvailability(scheduleId, TicketAvailabilityRequest.of(purchaseTicketCount));
	}
}

File: apis/src/main/kotlin/com/beat/apis/schedule/facade/ScheduleFacade.kt
package com.beat.apis.schedule.facade

import com.beat.apis.schedule.api.response.TicketAvailabilityResponse
import com.beat.apis.schedule.application.query.ScheduleQueryService
import org.springframework.stereotype.Service

@Service
class ScheduleFacade(
    private val scheduleQueryService: ScheduleQueryService,
) {
    fun findTicketAvailability(scheduleId: Long, purchaseTicketCount: Int): TicketAvailabilityResponse =
        TicketAvailabilityResponse.from(
            scheduleQueryService.findTicketAvailability(
                scheduleId,
                purchaseTicketCount,
            ),
        )
}
